### ISSUE TITLE
Add cache memory limit

### ISSUE THREAD

**Issue by lnp5131** [qsniyg/maxurl#424]
Currently it has a duration to keep down memory usage, but there is no limit. On media heavy sites, this may start running into memory constraints.

**Comment by qsniyg** [comment]
I agree that would be helpful, but I'd have to figure out how to do that... I'm not sure if it's really possible to know the size in memory (as far as I know, javascript doesn't expose this information), but I could perhaps add a very rough estimate, based on the "content-length" header (which would correspond the size of the image/video file, but not the size of the media stored in ram).

I was also thinking of adding a "maximum items in cache" option, as that would probably fix _most_ use cases. Unless someone were to view many 10000x\* images at once, in _theory_ it should be able to scale decently well for memory usage.

**Comment by lnp5131** [comment]
I also think the "maximum items in cache" option would fix most use cases. It seems like there are other ways to measure the memory size of objects, but it seems to suffer some instability as the object becomes large.

**Comment by qsniyg** [comment]
The maximum items in cache option has been implemented now: https://github.com/qsniyg/maxurl/issues/419

I'll still keep this one open though, it's a good idea in theory, just currently have no idea how to actually implement it haha.
